job/src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -590,7 +567,11 @@ public boolean hasntStartedYet() {
             for (SCMCheckout co : checkouts(null)) {
                 if (co.changelogFile != null && co.changelogFile.isFile()) {
                     try {
-                        changeSets.add(co.scm.createChangeLogParser().parse(this, co.scm.getEffectiveBrowser(), co.changelogFile));
+                        ChangeLogSet<? extends ChangeLogSet.Entry> changeLogSet =
+                                co.scm.createChangeLogParser().parse(this, co.scm.getEffectiveBrowser(), co.changelogFile);
+                        if (!changeLogSet.isEmptySet()) {
```

Comment:
BTW full test coverage in this case would require use of `RestartableJenkinsRule`, but I am not about to insist on it since the patch is pretty simple.